core/src/Kokkos_Concepts.hpp

```diff
@@ -120,6 +120,20 @@ struct LaunchBounds {
   static constexpr unsigned int minBperSM{minB};
 };
 
+template <int size>
+struct SubGroupSize {
```

Comment:
Do these make sense for all backends ?

Reply:
No, it's for SYCL only, just as `LaunchBounds` is for Cuda and HIP only.

Reply:
Also, this pull request introduces `RegisterFileSize` not `SubGroupSize`.

Reply:
OK, but they are defined at global level, and visible always. Is it intended ? (just a question for my understanding)

Reply:
OK. Food for discussion then. There might be a similar `CUDA` option called [__maxnreg__](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#maximum-number-of-registers-per-thread).

IMO anything in `Kokkos_Concepts.hpp` should have a broad adoption throughout the majority of the supported backends, otherwise it does not feel like a concepts but rather a backend-specific tuning property.